shadows/framework/src/main/java/org/robolectric/shadows/ShadowDeviceConfig.java

```diff
@@ -13,4 +16,20 @@ protected static String getProperty(String namespace, String name) {
     // avoid call to Settings.Config
     return null;
   }
+
+  @Resetter
+  public static void reset() {
+    Object lock = ReflectionHelpers.getStaticField(DeviceConfig.class, "sLock");
+    //noinspection SynchronizationOnLocalVariableOrMethodParameter
+    synchronized (lock) {
+      Map singleListeners = ReflectionHelpers.getStaticField(DeviceConfig.class, "sSingleListeners");
+      singleListeners.clear();
+
+      Map listeners = ReflectionHelpers.getStaticField(DeviceConfig.class, "sListeners");
+      listeners.clear();
+
+      Map namespaces = ReflectionHelpers.getStaticField(DeviceConfig.class, "sNamespaces");
```

Comment:
I don't think sNamespaces is safe to clear. Internally, when I do this, most Robolectric tests fail with the error:

```
this method should only be called by Robolectric
java.lang.IllegalStateException: 
	at org.robolectric.shadows.ShadowDisplayManager.configureDefaultDisplay(ShadowDisplayManager.java:69)
	at org.robolectric.android.Bootstrap.setUpDisplay(Bootstrap.java:18)
	at org.robolectric.android.internal.AndroidTestEnvironment.installAndCreateApplication(AndroidTestEnvironment.java:219)
	at org.robolectric.android.internal.AndroidTestEnvironment.setUpApplicationState(AndroidTestEnvironment.java:171)
	at org.robolectric.RobolectricTestRunner.beforeTest(RobolectricTestRunner.java:319)
	at org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$0(SandboxTestRunner.java:254)
	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:89)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Weird how that doesn't happen in CircleCI. 

However, if sNamespaces is not cleared, everything seems to work.

Reply:
Actually, the real error message was:

```
java.lang.RuntimeException: java.lang.NoSuchFieldException: sSingleListeners
	at org.robolectric.util.ReflectionHelpers.getStaticField(ReflectionHelpers.java:200)
	at org.robolectric.shadows.ShadowDeviceConfig.reset(ShadowDeviceConfig.java:26)
	at org.robolectric.Shadows.reset(Shadows.java:2341)
	at org.robolectric.android.internal.AndroidTestEnvironment.resetState(AndroidTestEnvironment.java:518)
	at org.robolectric.RobolectricTestRunner.lambda$finallyAfterTest$0(RobolectricTestRunner.java:349)
	at org.robolectric.util.PerfStatsCollector.measure(PerfStatsCollector.java:75)
	at org.robolectric.RobolectricTestRunner.finallyAfterTest(RobolectricTestRunner.java:347)
	at org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$0(SandboxTestRunner.java:272)
	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:89)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.NoSuchFieldException: sSingleListeners
	at java.base/java.lang.Class.getDeclaredField(Class.java:2411)
	at org.robolectric.util.ReflectionHelpers.getStaticField(ReflectionHelpers.java:198)
	... 12 more
```

sSingleListeners is only available in API 29, so that should only be cleared with API 29. I think sNamespaces is fine to clear.